tests/integration/components/ui-message-test.js

```diff
@@ -0,0 +1,315 @@
+    closeable=true
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $closeableIcon = $component.children('i');
+
+  assert.strictEqual($closeableIcon.hasClass('close'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual($closeableIcon.hasClass('icon'), true, 'Component\'s wrapper has css class');
+});
+
+test('caption & massage renders properly', function(assert) {
+  assert.expect(3);
+
+  // Render component.
+  this.render(hbs`{{ui-message
+    class=class
+    caption='My caption'
+    message='My message'
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $captionText = $component.children('div');
+  let $massageText = $component.children('p');
+
+  assert.strictEqual($captionText.hasClass('header'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual(Ember.$.trim($captionText.text()), 'My caption', 'Text component\'s caption right');
+  assert.strictEqual(Ember.$.trim($massageText.text()), 'My message', 'Text component\'s caption right');
+});
+
+test('icon renders properly', function(assert) {
+  assert.expect(7);
+
+  // Render component.
+  this.render(hbs`{{ui-message
+    class=class
+    icon='icon paw'
+    caption='My caption'
+    message='My message'
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $messageIcon = $component.children('i');
+  let $captionDiv = $component.children('div');
+  let $captionText = $captionDiv.children('div');
+  let $massageText = $captionDiv.children('p');
+
+  assert.strictEqual($component.hasClass('icon'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual($messageIcon.hasClass('paw'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual($messageIcon.hasClass('icon'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual($captionDiv.hasClass('content'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual($captionText.hasClass('header'), true, 'Component\'s wrapper has css class');
+  assert.strictEqual(Ember.$.trim($captionText.text()), 'My caption', 'Text component\'s caption right');
+  assert.strictEqual(Ember.$.trim($massageText.text()), 'My message', 'Text component\'s caption right');
+});
+
+test('component sends \'onHide\' action', function(assert) {
+  assert.expect(3);
+
+  let messageClose = false;
+  this.set('actions.onClose', () => {
+    messageClose = true;
+  });
+
+  // Render component.
+  this.render(hbs`{{ui-message
+    class=class
+    closeable=true
+    onHide=(action \"onClose\")
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $closeableIcon = $component.children('i');
+
+  // The component is visible.
+  assert.strictEqual($component.hasClass('hidden'), false, 'Component\'s wrapper has css class \'hidden\'');
+
+  // The component is hidden by the Close button.
+  Ember.run(() => {
+    $closeableIcon.click();
+    setTimeout(() => {
```

Comment:
Call to setTimeout causes asynchronous execution, but there is no call to let done = asert.async() & no following call to done() after all assertions. Add them please & add desired timeout in milliseconds (50 for example) setTimeout(() => {...}, 50).